### PR TITLE
Call dismissAllModals completion handler after the dismissal has actually completed

### LINKED ISSUE #5045: [V2] dismissAllModals doesn't await
### Issue Description

Similarly to #3733, it's not possible to await dismissAllModals() then trying to use showModal() after it

### Steps to Reproduce / Code Snippets / Screenshots

```
await Navigation.dismissAllModals(); 
Navigation.showModal('modal');
```

---
### Environment
* React Native Navigation version: 2.8
* React Native version: 0.57
* Platform(s) (iOS, Android, or both?): tested on iOS
* Device info (Simulator/Device? OS version? Debug/Release?): iOS 11.4 (Simulator)

### PR DESCRIPTION
**Pull request by danilobuerger**
This moves the call to the completion handler to a later time. Instead of relying on the completion of the CATransaction, we are now waiting until [UIViewController dismissViewControllerAnimated:completion:] calls back

Fixes #5045, #5552

### CODE REVIEW

File: lib/ios/RNNCommandsHandler.m [hunk 1]
```diff
@@ -312,10 +312,11 @@ - (void)dismissAllModals:(NSDictionary *)mergeOptions commandId:(NSString*)comma
 	[CATransaction begin];
 	[CATransaction setCompletionBlock:^{
 		[self->_eventEmitter sendOnNavigationCommandCompletion:dismissAllModals commandId:commandId];
```
[the remark this answers is not in the document]
Comment (danilobuerger) [reply lib/ios/RNNCommandsHandler.m:314]: @yogevbd Sure, I just followed the logic from `dismissModal`. If we move the event emitter here, we should also do it in `dismissModal`, yes?

[the remark this answers is not in the document]
Comment (yogevbd) [reply lib/ios/RNNCommandsHandler.m:314]: @danilobuerger Yes 😆💪

[the remark this answers is not in the document]
Comment (danilobuerger) [reply lib/ios/RNNCommandsHandler.m:314]: @yogevbd resolved

**Review by yogevbd — APPROVED** [review]

**Comment by zabojad** [comment]
There is an issue with this change. If there is no modal shown, the call to `dismissAllModals` does not resolve at all, neither in  `then` nor in `catch`...

**Pull request merged**